parm/config/gefs/config.base

```diff
@@ -98,7 +98,7 @@ export PSLOT="@PSLOT@"
 export EXPDIR="@EXPDIR@/${PSLOT}"
 export ROTDIR="@COMROOT@/${PSLOT}"
 
-export DATAROOT="${STMP}/RUNDIRS/${PSLOT}"  # TODO: set via prod_envir in Ops
+export DATAROOT="${STMP}/RUNDIRS/${PSLOT}/${RUN}.${PDY}${cyc}"  # TODO: set via prod_envir in Ops
```

Comment:
The workflow sets the environmental variable `DATAROOT` based on this value in [tasks.py](https://github.com/NOAA-EMC/global-workflow/blob/8998ec7b74123e953b97a93fa14cc78d471a1aee/workflow/rocoto/tasks.py#L68).  Should this be removed from tasks.py so that `DATAROOT` is set when sourcing `config.base` at execution time?  I'm not sure if `DATAROOT` is needed before then or not.

Reply:
It looks like `DATAROOT` is needed before sourcing the configs.  `jjob_header.sh` attempts to create the `DATA` directory before sourcing configs:

https://github.com/NOAA-EMC/global-workflow/blob/8998ec7b74123e953b97a93fa14cc78d471a1aee/ush/jjob_header.sh#L69-L104

The order would need to be changed if `DATAROOT` is to be removed from the .xml.

Reply:
I checked this order in some jobs from current ops, and found that the configs are sourced first, then the `DATA` directory is created and finally `setPDY.sh` is sourced.
I don't see a reason why we cannot do that here as well.  I will make the change in this PR, but we will wait for confirmation from @WalterKolczynski-NOAA

Reply:
I remember the history on why `DATAROOT` is being set from `tasks.py`.  This was done to emulate ecFlow setup.  `DATAROOT` is a variable set via the module `prod_envir` on WCOSS2.  When we eventually do a code handoff, there will not be a `DATAROOT` in `config.base`.  Its existence in `config.base` is purely for development purposes.

Reply:
Could another option be to remove `DATAROOT` from config.base and just let rocoto declare it?  It would have to be templated in tasks.py:

```python
        envar_dict = {'RUN_ENVIR': self._base.get('RUN_ENVIR', 'emc'),
                      'HOMEgfs': self.HOMEgfs,
                      'EXPDIR': self._base.get('EXPDIR'),
                      'NET': self._base.get('NET'),
                      'CDUMP': self.cdump,
                      'RUN': self.cdump,
                      'CDATE': '<cyclestr>@Y@m@d@H</cyclestr>',
                      'PDY': '<cyclestr>@Y@m@d</cyclestr>',
                      'cyc': '<cyclestr>@H</cyclestr>',
                      'COMROOT': self._base.get('COMROOT'),
                      'DATAROOT': self._base.get('STMP') + "/RUNDIRS/" + self._base.get('PSLOT') + "/" + self.cdump + ".<cyclestr>@Y@m@d@H</cyclestr>"}
```

Reply:
Doing that will make it difficult to change the location of RUNDIRS per experiment from the same clone of global-workflow.

Reply:
accepted suggestion.